### ISSUE TITLE
Document how to write a config.py that loads config from the environment

### ISSUE THREAD

**Issue by davidism** [pallets/flask#2329]
On Heroku, we can't add a local config `instance/config.py` that's not part of the repo. Instead, they support configuration through env vars. Document this pattern.

`myapp/config.py`:

~~~python
MAIL_USERNAME = os.environ.get('FLASK_MAIL_USERNAME', 'default')
LDAP_PORT = int(os.environ.get('FLASK_LDAP_PORT', 636))
~~~

**Comment by kennethreitz** [comment]
@davidism where in the docs should this go?

**Comment by davidism** [comment]
@kennethreitz this can be a section in `docs/config.rst`.

**Comment by tristanfisher** [comment]
I did a little write-up that I think works as a starting point.  I intentionally didn't go into configuration file composition (e.g. `config_prod.py` where a secret key is required and debug is statically set to false) as that sounds like the role of the following "best practices" section and I opted for duplication of text from the previous "configuration from files" for reasons of anchor links.

Let me know if my branch name needs to change (slipped my mind to create a more descriptive branch -- will follow what seems like a convention in other PRs from now on) and I'll re-open the PR.
